apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoBlockEditor/Settings/MuxSubtitles/MuxSubtitleSwitch.tsx

```diff
@@ -0,0 +1,181 @@
+import { gql, useQuery } from '@apollo/client'
+import Stack from '@mui/material/Stack'
+import Switch from '@mui/material/Switch'
+import Typography from '@mui/material/Typography'
+import { useTranslation } from 'next-i18next'
+import { ReactElement, useEffect, useState } from 'react'
+
+import { useEditor } from '@core/journeys/ui/EditorProvider'
+
+import { BlockFields_VideoBlock as VideoBlock } from '../../../../../../../../../__generated__/BlockFields'
+import { GetMyGeneratedMuxSubtitleTrack } from '../../../../../../../../../__generated__/GetMyGeneratedMuxSubtitleTrack'
+import { useValidateMuxLanguage } from '../../../../../../../../libs/useValidateMuxLanguage'
+
+export const GET_MY_GENERATED_MUX_SUBTITLE_TRACK = gql`
+  query GetMyGeneratedMuxSubtitleTrack($muxVideoId: ID!, $bcp47: String!) {
+    getMyGeneratedMuxSubtitleTrack(muxVideoId: $muxVideoId, bcp47: $bcp47) {
+      __typename
+      ... on QueryGetMyGeneratedMuxSubtitleTrackSuccess {
+        data {
+          id
+          status
+        }
+      }
+      ... on Error {
+        message
+      }
+    }
+  }
+`
+
+interface MuxSubtitleSwitchProps {
+  videoBlockId: string | null
+  muxVideoId: string | null
+  journeyLanguageCode: string | null | undefined
+  onChange: (showGeneratedSubtitles: boolean) => Promise<void>
+}
+
+export function MuxSubtitleSwitch({
+  videoBlockId,
+  muxVideoId,
+  journeyLanguageCode,
+  onChange
+}: MuxSubtitleSwitchProps): ReactElement {
+  const { t } = useTranslation('apps-journeys-admin')
+  const {
+    state: { selectedBlock }
+  } = useEditor()
+
+  const videoBlock =
+    selectedBlock?.__typename === 'VideoBlock' ? selectedBlock : undefined
+
+  const isValidLanguage = useValidateMuxLanguage(journeyLanguageCode)
+  const [toggleChecked, setToggleChecked] = useState(
+    videoBlock?.showGeneratedSubtitles ?? false
+  )
+  const [updating, setUpdating] = useState(false)
+
+  // Query subtitle track status
+  const {
+    data: subtitleTrackData,
+    error: subtitleTrackError,
+    startPolling,
+    stopPolling
+  } = useQuery<GetMyGeneratedMuxSubtitleTrack>(
+    GET_MY_GENERATED_MUX_SUBTITLE_TRACK,
+    {
+      variables: {
+        muxVideoId: muxVideoId ?? '',
+        bcp47: journeyLanguageCode ?? ''
+      },
+      skip:
+        !isValidLanguage || muxVideoId == null || journeyLanguageCode == null
```

Comment:
```suggestion
        !isValidLanguage || muxVideoId == null || journeyLanguageCode == null ||         videoBlock?.showGeneratedSubtitles == true ||
        videoBlock?.showGeneratedSubtitles == false
```

can you please try adding this in. This is meant to stop fetching the sub titles if they have already been generated and the status has either been verified that it is no longer processing.